images_config/kickstart/kickstart-robot.sh

```diff
@@ -0,0 +1,465 @@
+#!/bin/bash
+
+function logFile {
+    touch /root/kickstart_log
+    $1 >> /root/kickstart_log
+}
+
+function printHeader {
+    echo "#############################################"
+    echo "Execute $1"
+    echo "---------------------------------------------"
+}
+
+function SetLocalAptCacher {
+    printHeader "SetLocalAptCacher"
+    unset http_proxy
+    touch /etc/apt/apt.conf.d/01proxy
+
+    SERVERNAME=$(echo ${HOSTNAME%-*}-b1)
+    if grep -q 'Acquire::http { Proxy "http://10.0.1.1:3142"; };' /etc/apt/apt.conf.d/01proxy ; then
+        echo "Proxy already in /etc/apt/apt.conf.d/01proxy, skipping SetLocalAptCacher"
+    fi 
+    if grep -q 'Acquire::http { Proxy "http://'$SERVERNAME':3142"; };' /etc/apt/apt.conf.d/01proxy ; then
+        rm /etc/apt/apt.conf.d/01proxy
+        touch /etc/apt/apt.conf.d/01proxy
+        echo 'Acquire::http { Proxy "http://10.0.1.1:3142"; };' >>  /etc/apt/apt.conf.d/01proxy
+    else
+        echo 'Acquire::http { Proxy "http://10.0.1.1:3142"; };' >>  /etc/apt/apt.conf.d/01proxy
+    fi
+}
+
+# kernel 4.13 supports realsense no need to patch kernel.
+# package needs to be installed manually for now
+function BlockRealsensePackage {
+    if [ "$DISTRO" == "xenial" ]; then
+        apt-mark hold ros-kinetic-librealsense
+        apt-mark hold ros-kinetic-realsense-camera
+    fi
+}
+
+function AddGnomePPA {
+    printHeader "AddGnomePPA"
+    add-apt-repository ppa:gnome3-team/gnome3 -y
+    apt-get update
+}
+
+function EnableAptSources {
+    printHeader "EnableAptSources"
+    sed -i "s/# deb-src/deb-src/g" /etc/apt/sources.list
+    apt-get update
+}
+
+function UpgradeAptPackages {
+    printHeader "UpgradeAptPackages"
+    apt-get update
+    apt-get upgrade -y
+    apt-get dist-upgrade -y
+}
+
+# TODO: fxm
+function UpgradeKernel {
+    printHeader "UpgradeKernel"
+    if [ "$DISTRO" == "trusty" ]; then
+        apt-get install --install-recommends linux-generic-lts-xenial -y --allow
+    fi
+}
+
+function InstallHWEnableStacks {
```

Comment:
if we use HW-enabled seed for 14.04 (like we do for 16.04), this step is not required here anymore